src/checkers/htmlchecker.py

```diff
@@ -109,6 +109,24 @@ def check(self, external_data):
 
         self._update_version(external_data, latest_version, abs_url)
 
+    @staticmethod
+    def _substitute_placeholders(template_string, version):
+        version_list = LooseVersion(version).version
+        number_of_fields = len(version_list)
+        template_obj = Template(template_string)
+
+        replacements = {"version": version, "major": version_list[0]}
+
+        for i in range(number_of_fields):
+            replacements["version" + str(i)] = str(version_list[i])
+
+        if number_of_fields > 1:
+            template = template_obj.safe_substitute({"minor": str(version_list[1])})
+            if number_of_fields > 2:
+                template = template_obj.safe_substitute({"patch": str(version_list[1])})
```

Comment:
`template` variable goes nowhere, substituting the template here is pointless. Assign `replacements` entries instead.